packages/devtools-components/src/tests/tree.js

```diff
@@ -195,6 +203,159 @@ describe("Tree", () => {
+    expect(wrapper.find(".focused").prop("id")).toBe("key-A");
+    expect(wrapper.find(".active").exists()).toBe(false);
+  });
+
+  it("active item - renders as expected when tree blurs", () => {
+    const wrapper = mountTree({
+      expanded: new Set("ABCDEFGHIJKLMNO".split("")),
+      focused: "G",
+      active: "G"
+    });
+    expect(formatTree(wrapper)).toMatchSnapshot();
+    expect(wrapper.find(".active").prop("id")).toBe("key-G");
+
+    wrapper.simulate("blur");
+    expect(formatTree(wrapper)).toMatchSnapshot();
+    expect(wrapper.find(".active").exists()).toBe(false);
+  });
+
+  it("active item - renders as expected when moving away with keyboard", () => {
+    const wrapper = mountTree({
+      expanded: new Set("ABCDEFGHIJKLMNO".split("")),
+      focused: "L",
+      active: "L"
+    });
+    expect(formatTree(wrapper)).toMatchSnapshot();
+    expect(wrapper.find(".active").prop("id")).toBe("key-L");
+
+    simulateKeyDown(wrapper, "ArrowUp");
+    expect(wrapper.find(".active").exists()).toBe(false);
+  });
+
+  it("active item - renders as expected when using keyboard and Enter", () => {
+    const wrapper = mountTree({
+      expanded: new Set("ABCDEFGHIJKLMNO".split("")),
+      focused: "L"
+    });
+    wrapper.getDOMNode().focus();
+    expect(formatTree(wrapper)).toMatchSnapshot();
+    expect(wrapper.find(".active").exists()).toBe(false);
+
+    simulateKeyDown(wrapper, "Enter");
+    expect(wrapper.find(".active").prop("id")).toBe("key-L");
+
+    expect(wrapper.getDOMNode().ownerDocument.activeElement).toBe(
+      wrapper.getDOMNode()
+    );
+
+    simulateKeyDown(wrapper, "Escape");
+    expect(wrapper.find(".active").exists()).toBe(false);
+    expect(wrapper.getDOMNode().ownerDocument.activeElement).toBe(
+      wrapper.getDOMNode()
+    );
+  });
+
+  it("active item - renders as expected when using keyboard and Space", () => {
+    const wrapper = mountTree({
+      expanded: new Set("ABCDEFGHIJKLMNO".split("")),
+      focused: "L"
+    });
+    wrapper.getDOMNode().focus();
+    expect(formatTree(wrapper)).toMatchSnapshot();
+    expect(wrapper.find(".active").exists()).toBe(false);
+
+    simulateKeyDown(wrapper, " ");
+    expect(wrapper.find(".active").prop("id")).toBe("key-L");
+
+    simulateKeyDown(wrapper, "Escape");
+    expect(wrapper.find(".active").exists()).toBe(false);
+  });
+
+  it("active item - focus is inside the tree node when possible", () => {
+    const wrapper = mountTree({
+      expanded: new Set("ABCDEFGHIJKLMNO".split("")),
+      focused: "L",
+      renderItem: renderItemWithFocusableContent
+    });
+    wrapper.getDOMNode().focus();
+    expect(formatTree(wrapper)).toMatchSnapshot();
+    expect(wrapper.find(".active").exists()).toBe(false);
+    expect(wrapper.getDOMNode().ownerDocument.activeElement).toBe(
+      wrapper.getDOMNode()
+    );
+
+    simulateKeyDown(wrapper, "Enter");
+    expect(wrapper.find(".active").prop("id")).toBe("key-L");
```

Comment:
maybe we could have a snapshot check here as well `expect(formatTree(wrapper)).toMatchSnapshot();`